calibration_assistant.cpp

```diff
@@ -0,0 +1,1250 @@
+    m_pSlewBtn->Bind(wxEVT_COMMAND_BUTTON_CLICKED, &CalibrationAssistant::OnSlew, this);
+    m_pCalibrateBtn = new wxButton(this, wxID_ANY, _("Calibrate"));
+    m_pCalibrateBtn->SetToolTip(_("Starts the PHD2 calibration.  Calibration Assistant window will remain open to monitor and assess results"));
+    m_pCalibrateBtn->Bind(wxEVT_COMMAND_BUTTON_CLICKED, &CalibrationAssistant::OnCalibrate, this);
+    wxButton* pCancelBtn = new wxButton(this, wxID_ANY, _("Cancel"));
+    pCancelBtn->SetToolTip(_("Closes the Calibration Assistant window.  Any calibration currently underway will continue."));
+    pCancelBtn->Bind(wxEVT_COMMAND_BUTTON_CLICKED, &CalibrationAssistant::OnCancel, this);
+
+    btnSizer->Add(m_pSlewBtn, wxSizerFlags().Border(wxALL, 20));
+    btnSizer->Add(m_pCalibrateBtn, wxSizerFlags().Border(wxALL, 20));
+    btnSizer->Add(pCancelBtn, wxSizerFlags().Border(wxALL, 20));
+
+    m_pExplainBtn = new wxButton(this, wxID_ANY, _("Explain"));
+    m_pExplainBtn->SetToolTip(_("Shows additional information about any calibration result that is less than 'good'"));
+    m_pExplainBtn->Bind(wxEVT_COMMAND_BUTTON_CLICKED, &CalibrationAssistant::OnExplain, this);
+    wxBoxSizer* vSizer = new wxBoxSizer(wxVERTICAL);
+    vSizer->Add(m_pExplanation, wxSizerFlags().Center().Border(wxTop, 5));
+    vSizer->Add(currSizer, wxSizerFlags().Center().Border(wxALL, 20));
+    vSizer->Add(tgtSizer, wxSizerFlags().Center());
+    vSizer->Add(midBtnSizer, wxSizerFlags().Center().Border(wxTOP, 5));
+    vSizer->Add(m_pMessage, wxSizerFlags().Center().Border(wxTOP, 15));
+    vSizer->Add(m_pExplainBtn, wxSizerFlags().Center().Border(wxALL, 10));
+    vSizer->Add(m_pWarning, wxSizerFlags().Center().Border(wxTOP, 15));
+    vSizer->Add(btnSizer, wxSizerFlags().Center().Border(wxTOP, 15));
+
+    m_pTimer = new wxTimer(this, wxID_ANY);     // asynch updates to current position fields
+    m_pTimer->SetOwner(this);
+    this->Connect(wxEVT_TIMER, wxTimerEventHandler(CalibrationAssistant::OnTimer), NULL, this);
+    this->Bind(wxEVT_CLOSE_WINDOW, &CalibrationAssistant::OnClose, this);
+
+    InitializeUI(true);
+    m_pCurrOffset->Enable(false);
+    m_pCurrDec->Enable(false);
+    m_pCurrWest->Enable(false);
+    m_pCurrEast->Enable(false);
+    SetAutoLayout(true);
+    SetSizerAndFit(vSizer);
+    m_pExplainBtn->Enable(false);
+}
+
+CalibrationAssistant::~CalibrationAssistant()
+{
+    delete m_pTimer;
+    pFrame->pCalibrationAssistant = NULL;
+}
+
+void CalibrationAssistant::GetCustomLocation(int* PrefHA, int* PrefDec, bool* SingleSide, bool* UsingDefaults) const
+{
+    *PrefHA = pConfig->Profile.GetInt ("/scope/CalSlew/TgtHA", defBestOffset);
+    *PrefDec = pConfig->Profile.GetInt("/scope/CalSlew/TgtDec", defBestDec);
+    *SingleSide = pConfig->Profile.GetBoolean("/scope/CalSlew/SingleSide", false);
+    *UsingDefaults = (*PrefDec == defBestDec && *PrefHA == defBestOffset && !*SingleSide);
+}
+
+void CalibrationAssistant::PerformSanityChecks(void)

```

Comment:
optional: you could eliminate much of the indentation to improve readability by adding some early returns:
```
    if (!pPointingSource || !pPointingSource->IsConnected())
        return;

    double raSpd, decSpd;
    bool error = pPointingSource->GetGuideRates(&raSpd, &decSpd);
    if (error)
        return;

    if (!pPointingSource->ValidGuideRates(raSpd, decSpd))
        return;

    ... and so forth ...